hamilton/src/hybrid_planner.cpp

```diff
@@ -0,0 +1,576 @@
+		new descartes_trajectory::JointTrajectoryPt(start_pose));
+
+		// Creating End JointTrajectoryPt from end joint pose
+		descartes_core::TrajectoryPtPtr end_joint_point = descartes_core::TrajectoryPtPtr(
+		new descartes_trajectory::JointTrajectoryPt(end_pose));
+
+		// Modifying start and end of the trajectory.
+		descartes_traj_input_[0] = start_joint_point;
+		descartes_traj_input_[descartes_traj_input_.size() - 1 ] = end_joint_point;
+		}
+		else
+		{
+		ROS_ERROR_STREAM("Failed to find closest joint pose to seed pose at the start or end of trajectory");
+		exit(-1);
+		}
+
+		//TOCHECK Should the planners be initialized/modified in here instead of init()
+		//as the RobotPtr might have changed in its internal parameters
+
+		if(config_.descartes_planner_type == "dense")
+		{
+			bool succeeded = dense_planner_.planPath(descartes_traj_input_);
+			if (succeeded)
+			{
+			ROS_INFO_STREAM("Valid path was found using the Descartes dense planner");
+			}
+			else
+			{
+			ROS_ERROR_STREAM("Could not solve for a valid path using the Descartes dense planner");
+			exit(-1);
+			}
+
+			succeeded = dense_planner_.getPath(descartes_traj_output_);
+
+			if(!succeeded || descartes_traj_output_.empty())
+			{
+			ROS_ERROR_STREAM("Failed to retrieve robot path");
+			}
+		}
+
+		if(config_.descartes_planner_type == "sparse")
+		{
+			bool succeeded = sparse_planner_.planPath(descartes_traj_input_);
+			if (succeeded)
+			{
+			ROS_INFO_STREAM("Valid path was found using the Descartes sparse planner");
+			}
+			else
+			{
+			ROS_ERROR_STREAM("Could not solve for a valid path using the Descartes sparse planner");
+			exit(-1);
+			}
+
+			succeeded = sparse_planner_.getPath(descartes_traj_output_);
+
+			if(!succeeded || descartes_traj_output_.empty())
+			{
+			ROS_ERROR_STREAM("Failed to retrieve robot path");
+			}
+		}
+		
+		ROS_INFO_STREAM("Task '"<<__FUNCTION__<<"' completed");
+	}
+
+	void HybridPlanner::runPath()
+	{
+		moveit_group_ptr_->setPlannerId(PLANNER_ID);//TOCHECK should be in init(). Also the config_ param or not doubt. Check header file. 
+		overall_robot_traj_.getRobotTrajectoryMsg(moveit_robot_traj_);
+
+		// creating goal joint pose to start of the path
+		// std::vector<double> seed_pose(robot_model_ptr_->getDOF());
+  		std::vector<double> start_pose;
+  		descartes_core::TrajectoryPtPtr first_point_ptr = path[0];
+		first_point_ptr->getNominalJointPose(config_.seed_pose,*descartes_robot_model_ptr_,start_pose);
+
+		// moving arm to joint goal
+		moveit_group_ptr_->setJointValueTarget(start_pose);
+		moveit_group_ptr_->setPlanningTime(config_.plan_time);
+		moveit_msgs::MoveItErrorCodes result = moveit_group_ptr_->move();
+		if(result.val != result.SUCCESS)
+		{
+		ROS_ERROR_STREAM("Move to start joint pose failed");
+		exit(-1);
+		}
+
+		// creating Moveit! joint trajectory from Descartes Trajectory
+		fromDescartesToMoveitTrajectory();
+
```

Comment:
Now this method transforms the `descartes_traj_output_` to the `trajectory_msgs::JointTrajectory& joint_traj_descartes_` member variable. 
So we need to combine and execute `joint_traj_descartes_` and `moveit_robot_traj` somehow. 
For this, I like this class http://docs.ros.org/hydro/api/moveit_core/html/classrobot__trajectory_1_1RobotTrajectory.html. It has addSuffixWayPoint and addPrefixWayPoint and so many useful methods. I just didn't know it existed 3 days before, and hence I was just stuck. 
By "opening up" one of the arrays, we could simply `InsertWayPoint()` using the times for different points? And I am afraid I have screwed up the times in the `fromDescartesToMoveitTrajectory` 
Please give your input here.